misc/FFNx.frag

```diff
@@ -72,28 +74,42 @@ void main()
     {
         if (isYUV)
         {
+            // BT601 TV-range YUV->RGB conversion
+            // (includes implict range conversion)
             const mat3 mpeg_rgb_transform = mat3(
-                vec3(+1.164, +1.164, +1.164),
-                vec3(+0.000, -0.392, +2.017),
-                vec3(+1.596, -0.813, +0.000)
+                vec3(+255.0 / 219.0, +255.0 / 219.0, +255.0 / 219.0),
+                vec3(+0.000, -25.75602 / 65.744 , +225.93 / 112.0),
+                vec3(+178.755 / 112.0, -53.447745 / 65.744 , +0.000)
             );
 
+            // BT601 full-range YUV->RGB conversion
             const mat3 jpeg_rgb_transform = mat3(
                 vec3(+1.000, +1.000, +1.000),
-                vec3(+0.000, -0.343, +1.765),
-                vec3(+1.400, -0.711, +0.000)
+                vec3(+0.000, -0.202008 / 0.587, +1.772),
+                vec3(+1.402, -0.419198 / 0.587, +0.000)
             );
 
             vec3 yuv = vec3(
-                texture2D(tex_0, v_texcoord0.xy).r - (1.0 / 16.0),
+                texture2D(tex_0, v_texcoord0.xy).r,
                 texture2D(tex_1, v_texcoord0.xy).r - 0.5,
                 texture2D(tex_2, v_texcoord0.xy).r - 0.5
             );
 
-            if (isFullRange) color.rgb = instMul(jpeg_rgb_transform, yuv);
-            else color.rgb = instMul(mpeg_rgb_transform, yuv);
+            if (isFullRange){
+                color.rgb = saturate(instMul(jpeg_rgb_transform, yuv));
+            }
+            else {
+                yuv.r = saturate(yuv.r - (1.0 / 16.0));
+                color.rgb = saturate(instMul(mpeg_rgb_transform, yuv));
+            }
 
-            color.rgb = toLinear(color.rgb);
+            
+            if (is170MGamma){
+                color.rgb = saturate(toLinearSMPTE170M(color.rgb));
+            }
+            else {
+                color.rgb = saturate(toLinear(color.rgb));
+            }

```

Comment:
I think this block of changes ( other than the matrix transformation which now let's say has more precision in it ) I think is not correct. Are you using `saturate` because you want to reach the 2.4 gamma out of the 2.2, or am I missing something here?